openmdao/core/system.py

```diff
@@ -937,6 +951,10 @@ def _apply_linear_jac(self, params, unknowns, dparams, dunknowns, dresids, mode)
 
         for (unknown, param), J in iteritems(self._jacobian_cache):
 
+            # Skip multiplying Jacobian on outscope vars
+            if self.rel_inputs and param not in self.rel_inputs:
```

Comment:
Might want to make self.rel_inputs a set rather than a list since you're checking containment in it repeatedly in a loop.